app/assets/javascripts/components/select_experience_domain.js

```diff
@@ -35,18 +41,28 @@ class SelectExperienceDomain extends React.PureComponent<Props, State> {
   }
 
   render() {
+    const options = this.getUnusedDomains().map(domain => <Option key={domain}>{domain}</Option>);
+    const notFoundContent = this.props.notFoundContent || "Not Found";
+    const additionalProps = {};
```

Comment:
Since `additionalProps` is only used for one component (in contrast to an earlier state, where you used it for two different branches with JSX), it doesn't add much value. You can directly inline it into the JSX below like that:

```
<Select
  ...
  onChange: this.props.onChange
```

It doesn't matter if, `this.props.onChange` is undefined or not (so no need for the if-switches).